feat(Task-7): As a User, I want a minimalistic StartPage.

jLagosCarrera opened the pull request.

jLagosCarrera committed:
feat(Task-1): Configure Webpack to our workflow

jLagosCarrera committed:
feat(Task-1): Add AngularJS hello world

jLagosCarrera committed:
feat(Task-1): Split config files for development and production

jLagosCarrera committed:
feat(Task-1): Add HtmlWebpackPlugin to serve index file

jLagosCarrera committed:
feat(Task-1): Add dist folder to gitignore

jLagosCarrera committed:
feat(Task-4): Create service for handling OWM request and use it on the component

jLagosCarrera committed:
Merge branch 'development' into feat/Task-4

jLagosCarrera committed:
Merge branch 'development' into feat/Task-4

jLagosCarrera committed:
refactor(Task-4): Structure all code to be more mantainable

jLagosCarrera committed:
feat(Task-4): Use of plugin to hide sensible information

jLagosCarrera committed:
feat(Task-5): Cleared all code from previous config tasks

jLagosCarrera committed:
Refactor(Task-5): Clarified a few lines

jLagosCarrera committed:
feat(Task-7): Start home page screen look and feel

jLagosCarrera committed:
feat(Task-7): Add input field and search button with validation

jLagosCarrera committed:
feat(Task-7): Created list for recent searches and managed save and load it on localStorage

jLagosCarrera committed:
refactor(Task-7): Restructured and adapted all code to ES6

jLagosCarrera committed:
feat(Task-7): Make API calls

jLagosCarrera committed:
Merge branch 'development' into feat/Task-7

jLagosCarrera committed:
feat(Task-7): Delete deprecated code

gbtec-ext-jorgevillar commented on src/app/app.module.js:
```diff
@@ -4,14 +4,24 @@ import angularmd from 'angular-material';
+import openWeatherMaps from './shared/services/openWeatherMaps';
+import latestSearchesUtil from './shared/services/latestSearchesUtil';
 
 export default angular.module('app', [uiRouter, angularanimate, angulararia, angularmessages, angularmd])
```
Use camel case for `angularanimate, angulararia, angularmessages, angularmd`

gbtec-ext-jorgevillar commented on src/app/shared/services/latestSearchesUtil.js:
```diff
@@ -0,0 +1,29 @@
+    };
+
+    addCity(city) {
+        if (city == undefined || city == null || city.trim() === '' || this.latestSearches.includes(city)) {
```
You can do this check `city == undefined || city == null || city.trim() === ''` with this `!city  || city.trim() === ''`

And be careful with this solution, Jennifer Null and her family doesn't like it. 
https://gizmodo.com/it-really-sucks-to-be-named-jennifer-null-1767159091

gbtec-ext-jorgevillar commented on src/app/shared/services/latestSearchesUtil.js:
```diff
@@ -0,0 +1,29 @@
+            return;
+        }
+
+        if (city === this.latestSearches[0]) {
```
isn't this check redundant with the `this.latestSearches.includes(city)` of the previous 'if'?

eloydotco commented on src/app/shared/services/latestSearchesUtil.js:
```diff
@@ -0,0 +1,29 @@
+export default class LatestSearchesUtil {
+    constructor($window, latestSearches) {
+        this.$window = $window;
```
`window` object is already available, it's not needed to inject it

eloydotco commented on src/app/shared/services/latestSearchesUtil.js:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    getLatestSearches() {
+        return this.latestSearches = JSON.parse(this.$window.localStorage.getItem('latestSearches')) || [];
```
`localStorage` is also available as "global object"

gbtec-ext-jorgevillar commented on src/app/shared/services/openWeatherMaps.js:
```diff
@@ -0,0 +1,42 @@
+            }
+        };
+
+        return this.fiveDayForecast = this.$http(request);
```
You can use the shortcut version:
https://code.angularjs.org/1.5.11/docs/api/ng/service/$http#get

gbtec-ext-jorgevillar commented on src/app/shared/services/openWeatherMaps.js:
```diff
@@ -0,0 +1,42 @@
+            }
+        };
+
+        return this.fiveDayForecast = this.$http(request);
```
Same as above

gbtec-ext-jorgevillar commented on src/app/shared/services/openWeatherMaps.js:
```diff
@@ -0,0 +1,42 @@
+        };
+
+        return this.fiveDayForecast = this.$http(request);
+    }
```
In this functions, do you want to return the response of the request? If that, you are failing :)

jLagosCarrera replied:
No, Im returning the promise

eloydotco replied:
I am with Jorge, you can return `this.$http.get(URL, params);` or first assign the value to `this.fiveDayForecast` and then return it, but not both.
Additionally, on the second option, `this.fiveDayForecast` will contain the promise, not the value, a better approach can be
````
return this.$http.get(URL, params).then(data => {
    this.fiveDayForecast = data; // maybe data need some transformation before assign it...
});

diegopf replied:
what about async / await? :)

jLagosCarrera committed:
fix(Task-7): Fix all requested changes

Use of $http.get shorthand, fixed city validation, used camelCase where it wasnt being used and access window and localstorage without injecting them

jLagosCarrera commented:
Requested changes were made

eloydotco commented on src/app/app.module.js:
```diff
@@ -1,17 +1,27 @@
+import angularAnimate from 'angular-animate';
+import angularAria from 'angular-aria';
+import angularMessages from 'angular-messages';
+import 'angular-material/angular-material.scss';
```
it's a good practice add and empty line between different imports blocks (I mean , by type, libraries, styles, services, etc), just for readability

eloydotco commented on src/app/shared/services/openWeatherMaps.js:
```diff
@@ -0,0 +1,38 @@
+            }
+        };
+
+        //return daily forecast promise
```
I think here there is something missing...

jLagosCarrera committed:
fix(Task-7): Fix OWM service methods

jLagosCarrera commented:
Requested changes were made

diegopf commented on src/app/shared/services/openWeatherMaps.js:
```diff
@@ -0,0 +1,52 @@
+    };
+
+    getFiveDayForecastCity(city) {
+        const URL = 'https://api.openweathermap.org/data/2.5/forecast';
```
You should create a baseApi with 'https://api.openweathermap.org/data/2.5/' and build each endpoint url by concatenating the baseApi with the specific part

jLagosCarrera committed:
feat(Task-7): Style changes, not finished yet

jLagosCarrera committed:
feat(Task-7): Improve look and feel

jLagosCarrera committed:
feat(Task-7): Make API Call

jLagosCarrera committed:
feat(Task-7): Change .env location

jLagosCarrera committed:
feat(Task-7): Change config in webpack to allow netlify to work

jLagosCarrera committed:
feat(Task-7): Start converting promises to async functions

diegopf commented on src/app/shared/services/openWeatherMaps.js:
```diff
@@ -0,0 +1,60 @@
+        };
+
+        try {
+            return await this.$http.get(`${this.baseURL}/forecast`, params).then((response) => {
```
what? Don't mix await with resolving the promise via then.

jLagosCarrera replied:
Havent finished yet, sorry for commiting!

jLagosCarrera committed:
feat(Task-7): Remove old promises

jLagosCarrera committed:
feat(Task-7): Improved look and feel

jLagosCarrera committed:
feat(Task-7): Add padding to the icons

jLagosCarrera commented:
All requested changes are done

eloydotco commented on src/app/home-page/img/background.svg:
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="3840" height="2160"><path d="M2063,1007L2216,1002L2080,859Z" fill="#4aa4da" stroke="#4aa4da" stroke-width="1.51"/><path d="M2216,1002L2218,837L2080,859Z" fill="#48a8db" stroke="#48a8db" stroke-width="1.51"/><path d="M2063,1007L2229,1147L2216,1002Z" fill="#4ca2da" stroke="#4ca2da" stroke-width="1.51"/><path d="M2387,850L2236,702L2218,837Z" fill="#43ae ...
```
All assets/resources like images should be on a folder on the project root folder

jLagosCarrera committed:
feat(Task-7): Finished look and feel

jLagosCarrera committed:
fix(Task-7): Fix img path
